common/beerocks/bwl/common/key_value_parser.cpp

```diff
@@ -0,0 +1,225 @@
+
+#include <bwl/key_value_parser.h>
+
+#include <bcl/beerocks_state_machine.h>
+#include <bcl/beerocks_string_utils.h>
+#include <bcl/network/network_utils.h>
+
+#include <easylogging++.h>
+#include <sstream>
+
+using namespace bwl;
+
+namespace bwl {
+constexpr char EVENT_KEYLESS_PARAM_OPCODE[] = "_opcode";
+constexpr char EVENT_KEYLESS_PARAM_MAC[]    = "_mac";
+constexpr char EVENT_KEYLESS_PARAM_IFACE[]  = "_iface";
+} // namespace bwl
+
+//////////////////////////////////////////////////////////////////////////////
+////////////////////////// Local Module Functions ////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+/**
+ * @brief Finds first occurrence of two delimiters with string between them, on given string 'str':
+ * <'delim_near'><any_characters except delimiters><'delim_far'>
+ * 
+ * @param[in] str String to find the first occurrence of two delimiters with string between them.  
+ * @param[in] pos Postion of the first characters on 'str' to be considered in the search.
+ * @param[in] delim_near Near delimiter.
+ * @param[in] delim_far Near delimiter.
+ * @return std::string::size_type Postion of the first character that matches. If no matches is 
+ * found, returns string::npos.
+ */
+static std::string::size_type find_first_of_delimiter_pair(std::string &str,
+                                                           std::string::size_type pos,
+                                                           char delim_near, char delim_far)
+{
+    // Finds first occurrence on string 'str' of
+    // <'delim_near'><any_characters except delimiters><'delim_far'>
+    auto idx = str.find_first_of(delim_far, pos);
+    if (idx == std::string::npos) {
+        return idx;
+    }
+    idx = str.rfind(delim_near, idx);
+    return idx;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+/////////////////////////////// Implementation ///////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+void KeyValueParser::parse_line(std::stringstream &ss_in, std::list<char> delimiter_list,
```

Comment:
This could probably be
```suggestion
void KeyValueParser::parse_line(std::istream &in, std::list<char> delimiter_list,
```
there's no need to specialize to stringstream.

Reply:
Hmm yes, we can get a little better performance if we use `istringstream` (input only string stream).
`istream` is something we cannot use because it can't be constructed from c-string.
Anyway, replaced to `istringstream`.